engine/evalOp.cc

```diff
@@ -959,8 +978,14 @@ namespace minsky
                 if (to.idx()==-1 || to.xVector.empty())
                   to.setXVector(from1.xVector);
                 if (to.xVector==from1.xVector)
-                    for (size_t i=0; i<from1.numElements(); ++i)
-                      t->in1.push_back(i+from1.idx());
+                  { 
+					// For feature 47  
+                    if (from1.index.empty())   
+                      for (size_t i=0; i<from1.numDenseElements(); ++i)                          
+                        t->in1.push_back(i+from1.idx());
+                    else for (size_t i=0; i<from1.numSparseElements; ++i)
```

Comment:
This loop can be written more neatly using C++-11 for loops:
for (size_t i: from.index)  t->in1.push_back(i+from1.idx())

Also it highlights that numSparseElements is just equal to index.size() - you should read up on the dangers of unnormalised data

Reply:
I think this is what I'm really struggling with: that the normalized data structure in the xVector is not preserved upon import of sparse data files the way it is done at present. Storing the data in sparse format is easy. I can use two vectors, a vector of pairs, a map, etc. However, the really tricky part is serializing the data in the valuevectors for use in the Runge Kutta routines. The way I do it at present does not preserve the normalized data structure in the xVector.